### PR TITLE
[JBIDE-19949] Automatically import sources once an application is created

### PR DESCRIPTION
**Pull request by fbricon**
Signed-off-by: Fred Bricon <fbricon@gmail.com>

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java [hunk 1]
```diff
@@ -47,8 +50,28 @@ public ImportApplicationWizard() {
 		this.model = new ImportApplicationWizardModel();
 	}
 
+	public ImportApplicationWizard(Map<IProject, Collection<IBuildConfig>> projectsAndBuildConfigs) {
+		this();
+		if (projectsAndBuildConfigs != null && projectsAndBuildConfigs.size() == 1) {
+			Map.Entry<IProject, Collection<IBuildConfig>> entry = projectsAndBuildConfigs.entrySet().iterator().next();
+			IProject project = entry.getKey();
+			Connection connection = ConnectionsRegistryUtil.safeGetConnectionFor(project);
```
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java]: Presumably this should not be null but my only concern would be the case where the connection is not found; should we use the alternate getter?  Do we need to explicitly handle that case or display something to a user?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java [hunk 2]
```diff
@@ -47,8 +50,28 @@ public ImportApplicationWizard() {
...
+			setModelConnection(connection);
+			Collection<IBuildConfig> buildConfigs = entry.getValue();
+			if (buildConfigs.size() == 1) {
+				model.setSelectedItem(buildConfigs.iterator().next());
+			} else {
+				model.setSelectedItem(project);
```
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java]: What are we doing here?  Either we set to a buildconfig or project?  How is that manifested to a user? The buildconfig or the project is selected?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizard.java [hunk 3]
```diff
@@ -148,6 +168,25 @@ public void run() {
 		return success;
 	}
 
+	protected Map<IProject, Collection<IBuildConfig>> getBuildConfigs(Collection<IResource> resources) {
+		Map<IProject, Collection<IBuildConfig>> projects = new LinkedHashMap<>();
+		for (IResource resource : resources) {
+			if (resource instanceof IBuildConfig) {
```
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizard.java]: You can actually ask the kind of a resource via resource#getKind()  Not sure it really matters here though

**Comment by jcantrill** [comment]
My comments are just some items to note, not necessarily change requests.  LGTM

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java (hunk 2 above)
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java]: if 1 buildconfig is found, then we skip the buildconfig selection page and go directly to the git clone page.
If more buildconfigs are found, we select the parent project and display the buildconfig selection page (see addPages()). The user will have to choose what to import

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizard.java (hunk 3 above)
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizard.java]: Yeah I don't think that makes any difference here

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java (hunk 1 above)
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java]: If that blows here, then I don't know how to gracefully handle that. It's not any different that what is done in init().
I'd rather keep it as is, for the sake of simplicity.

**Pull request merged**